### PR TITLE
AVRO-1886: Add validation messages

### PR DESCRIPTION
**Pull request by MiroslavCsonka**
### Changes
- include error messages when validating a payload
- correct expectation in specs (http://ruby-doc.org/stdlib-2.1.0/libdoc/test/unit/rdoc/Test/Unit/Assertions.html#method-i-assert_equal)

Ticket https://issues.apache.org/jira/browse/AVRO-1886

### CODE REVIEW

**Comment by sldblog** [comment]
😍

File: lang/ruby/lib/avro/schema_validator.rb [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+          fail TypeMismatchError unless [true, false].include?(datum)
+        when :string, :bytes
+          fail TypeMismatchError unless datum.is_a?(String)
+        when :int
+          fail TypeMismatchError unless datum.is_a?(Fixnum) || datum.is_a?(Bignum)
+          range = (Schema::INT_MIN_VALUE..Schema::INT_MAX_VALUE)
+          result.add_error(path, "out of bound value #{datum}") unless range.cover?(datum)
+        when :long
+          fail TypeMismatchError unless datum.is_a?(Fixnum) || datum.is_a?(Bignum)
+          range = (Schema::LONG_MIN_VALUE..Schema::LONG_MAX_VALUE)
```
Comment (tjwp) [inline lang/ruby/lib/avro/schema_validator.rb]: Maybe this range should be a constant?

File: lang/ruby/lib/avro/schema_validator.rb [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+      def validate_union(expected_schema, datum, path, result)
+        if expected_schema.schemas.size == 1
+          validate_recursive(expected_schema.schemas.first, datum, path, result)
+          return
+        end
+        types_and_results = validate_possible_types(datum, expected_schema, path)
+        failures, successes = types_and_results.partition { |r| r[:result].failure? }
+        return if successes.any?
+        complex_types = [:array, :error, :map, :record, :request]
```
Comment (tjwp) [inline lang/ruby/lib/avro/schema_validator.rb]: Maybe this should be a constant?

**Comment by MiroslavCsonka** [comment]
@tjwp ♻️

**Comment by sldblog** [comment]
Is there anything we can do to push this pull request forward?

**Comment by rdblue** [comment]
@sldblog, if you want to review it, that would be helpful. Thanks!

**Pull request closed without merging**